app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -809,6 +809,10 @@ private void processCalibrationNoUI(final double glucosenumber, final double tim
                     alert.show();
                 }
                 // offer choice to enable auto-calibration mode if not already enabled on pratelimit
+            } else if (calibration_type.equals("never")) {
```

Comment:
Now, the code of the "never" branch is identical with the first three lines of the "auto" branch (except the log message). Is this intentional?

Code duplication can be avoided with:

```java
            } else if (calibration_type.equals("auto") || calibration_type.equals("never")) {
                Log.d(TAG, "Creating bloodtest record from cal input data");
                BloodTest.createFromCal(glucosenumber, timeoffset, "Manual Entry");
                GcmActivity.syncBloodTests();
                if ((calibration_type.equals("auto") && (!Pref.getBooleanDefaultFalse("bluetooth_meter_for_calibrations_auto"))
                        && (DexCollectionType.getDexCollectionType() != DexCollectionType.Follower)
                        && (JoH.pratelimit("ask_about_auto_calibration", 86400 * 30))) {
                    ...
                }
            } else {
```

Reply:
Yes, it's intentional.

Functionally, most of this makes no sense to me.  I don't even like automatic calibration.  
If someone wants to always use their entered test results for calibration, they can use automatic calibration.  So, I don't know why this whole thing has been organized this way.

When I use always, it does not add the reading on the main screen.  Why?  

Choosing auto, will ask you if you want to enable automatic calibration.  It's like we are trying to convince people to use automatic calibration.  Why?

If I answer no to enabling automatic calibration, after having auto enabled, I know that it adds the reading to the main screen.  That's what many people like to be able to do.  They can also tap on it and use it as calibration if they want.
Because I knew that did what was missing as an independent option, I created a new fork that just did that.
 
If this is merged, I will open a second step to remove the redundant (always use) fork.
If that is also merged, I would like to also remove the auto fork leaving only ask me always and never use.

Reply:
Thinking more about this, I don't think there should be any conditionals here at all.
Either the user enables automatic calibration or not.
If they haven't any readings through Bluetooth or manual should just be entered in the database and shown on screen as a BG test reading.

But, let's keep this as a small step.  In the process, someone may explain why this was done this way and clear up my confusion.